src/test/java/qna/domain/AnswerRepositoryTest.java

```diff
@@ -1,43 +1,67 @@
 package qna.domain;
 
 import static org.assertj.core.api.Assertions.*;
-import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.List;
 
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.annotation.DirtiesContext;
 
+@DirtiesContext
 @DataJpaTest
 class AnswerRepositoryTest {
 
 	@Autowired
 	private AnswerRepository repository;
 
-	private Answer save;
+	@Autowired
+	private UserRepository userRepository;
+
+	@Autowired
+	private QuestionRepository questionRepository;
+
+	private Answer answer;
+	private User savedUser;
+	private Question savedQuestion;
 
 	@BeforeEach
 	void setUp() {
-		save = repository.save(AnswerTest.A1);
+		savedUser = userRepository.save(new User("len", "password", "name", "email"));
+
+		Question question = new Question("title", "Contents");
+		savedQuestion = questionRepository.save(question);
+
+		answer = new Answer(savedUser, savedQuestion, "Answer Contents");
 	}
 
 	@Test
-	@DisplayName("정답을 저장된 후, ID를 갖는다")
-	void saveTest() {
-		assertAll(
-			() -> assertThat(save).isNotNull(),
-			() -> assertThat(save.getId()).isNotNull()
-		);
+	void findByIdAndDeletedFalseTest() {
+
+
+		Answer save = repository.save(answer);
+
+		Answer answer1 = repository.findByIdAndDeletedFalse(save.getId()).orElse(Answer.NONE);
```

Comment:
빈 객체를 반환하기 보다는 예외를 던져 처리해보면 좋겠네요 :)

```suggestion
		Answer answer1 = repository.findByIdAndDeletedFalse(save.getId()).orElseThrow(EntityNotFoundException::new);
```

Reply:
코드 수정하겠습니다 😀